lib/liquid/variable_lookup.rb

```diff
@@ -55,9 +55,11 @@ def evaluate(context)
           object = object.send(key).to_liquid
 
           # No key was present with the desired value and it wasn't one of the directly supported
-          # keywords either. The only thing we got left is to return nil
+          # keywords either. The only thing we got left is to return nil or
+          # raise an exception if `strict_variables` option is set to true
         else
-          return nil
+          return nil unless context.strict_variables
+          raise Liquid::UndefinedVariable, "undefined variable #{key} in #{object}"
```

Comment:
I see your point here about how this could be useful for developers, but I still disagree and at least in Shopify's use-case (where we don't necessarily trust the author of the template), I think exposing `object` is too dangerous.